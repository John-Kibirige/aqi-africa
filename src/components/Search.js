import PropTypes from 'prop-types';
import searchIcon from '../assets/search-icon.svg';

const Search = ({ nameState, handleOnChange }) => {
  const handleSubmit = (e) => {
    e.preventDefault();
  };
  return (
    <form
      className="search-form flex sm:mt-7 sm:mb-4 md:w-8/12 md:mx-auto mx-4"
      onSubmit={handleSubmit}
    >
      <input
        className="bg-gray-100 appearance-none border-2 border-pink-400 w-10/12 py-1 px-4 text-gray-700 leading-tight focus:outline-none focus:bg-white focus:border-pink-500 sm:p-2 md:p-3 text-xl"
        type="text"
        placeholder="Search by country"
        value={nameState}
        name={nameState}
        onChange={handleOnChange}
      />
      <button
        type="button"
        className="submit flex bg-pink-600 pl-1 align-middle w-2/12 justify-center"
      >
        <img src={searchIcon} alt="search" className="w-5 block md:w-6" />
      </button>
    </form>
  );
};

Search.propTypes = {
  nameState: PropTypes.string.isRequired,
  handleOnChange: PropTypes.func.isRequired,
};
export default Search;
