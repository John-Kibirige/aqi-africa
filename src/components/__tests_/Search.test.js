import React from 'react';
import renderer from 'react-test-renderer';
import { screen, render } from '@testing-library/react';
import Search from '../Search';

it('Renders when there are props passed to it', () => {
  const tree = renderer
    .create(<Search nameState="name state" handleOnChange={() => ''} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Render input field well ', () => {
  render(<Search nameState="name state" handleOnChange={() => ''} />);
  const inputEl = screen.getByPlaceholderText(/Search by country/i);
  expect(inputEl).toBeInTheDocument();
});
