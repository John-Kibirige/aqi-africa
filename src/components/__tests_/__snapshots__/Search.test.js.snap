// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders when there are props passed to it 1`] = `
<form
  className="search-form flex sm:mt-7 sm:mb-4 md:w-8/12 md:mx-auto mx-4"
  onSubmit={[Function]}
>
  <input
    className="bg-gray-100 appearance-none border-2 border-pink-400 w-10/12 py-1 px-4 text-gray-700 leading-tight focus:outline-none focus:bg-white focus:border-pink-500 sm:p-2 md:p-3 text-xl"
    name="name state"
    onChange={[Function]}
    placeholder="Search by country"
    type="text"
    value="name state"
  />
  <button
    className="submit flex bg-pink-600 pl-1 align-middle w-2/12 justify-center"
    type="button"
  >
    <img
      alt="search"
      className="w-5 block md:w-6"
      src="search-icon.svg"
    />
  </button>
</form>
`;
